flathunter/crawl_idealista.py

```diff
@@ -0,0 +1,59 @@
+"""Expose crawler for Idealista"""
+import logging
+import re
+
+import requests
+from bs4 import BeautifulSoup
+from flathunter.abstract_crawler import Crawler
+from random_user_agent.user_agent import UserAgent
+
+class CrawlIdealista(Crawler):
+    """Implementation of Crawler interface for Idealista"""
+
+    __log__ = logging.getLogger('flathunt')
+    URL_PATTERN = re.compile(r'https://www\.idealista\.it')
+
+    def __init__(self):
+        logging.getLogger("requests").setLevel(logging.WARNING)
+
+    # pylint: disable=too-many-locals
+    def extract_data(self, soup):
+        """Extracts all exposes from a provided Soup object"""
+        entries = list()
+
+        findings = soup.find_all('article', {"class": "item"})
+
+        base_url = 'https://www.idealista.it'
+        for row in findings:
+            title_row = row.find('a', {"class": "item-link"})
+            title = title_row.text.strip()
+            url = base_url + title_row['href']
+            picture_element = row.find('picture', {"class": "item-multimedia"})
+            if "no-pictures" not in picture_element.get("class"):
+                image = ""
+            else:
+                print(picture_element)
+                image = picture_element.find('img')['src']
+            detail_items = row.find_all("span", {"class": "item-detail"})
+            rooms = detail_items[0].text.strip()
+            size = detail_items[1].text.strip()
+            floor = detail_items[2].text.strip()
```

Comment:
I just ran this locally with the URL https://www.idealista.it/vendita-case/milano/garibaldi-porta-venezia/indipendenza-regina-giovanna/ , and I get an error:
```
File "/home/at/play/flathunter/flathunter/crawl_idealista.py" line 40, in extract_data
    floor = detail_items[2].text.strip()
IndexError: list index out of range
```

Reply:
Maybe add a check here for how big detail_items is, or just an exception handler for IndexError.

Reply:
The error was related to the fact that some fields are not present. Now it's fixed!